src/nifreeze/cli/run.py

```diff
@@ -92,12 +68,14 @@ def main(argv=None) -> None:
     )
 
     # Set the output filename to be the same as the input filename
-    output_filename: str = Path(args.input_file).name
+    output_filename = Path(Path(args.input_file).name).stem + ".nii.gz"
     output_path: Path = Path(args.output_dir) / output_filename
 
     # Save the DWI dataset to the output path
     if args.write_hdf5:
-        dataset.to_filename(output_path)
+        output_h5_filename = Path(Path(args.input_file).name).stem + ".h5"
+        output_h5_path: Path = Path(args.output_dir) / output_h5_filename
+        dataset.to_filename(output_h5_path)
```

Comment:
Although having an HDF5 file as input seems the best way for testing,
https://github.com/nipreps/nifreeze/blob/94f6d2708d15fbe85fb850094e956b63d9c047b9/src/nifreeze/cli/parser.py#L68

In the mid-term we will want to accept NIfTI files. So the above will need to change.

To be addressed in a separate PR.